src/parquet/util/rle-encoding.h

```diff
@@ -93,170 +94,171 @@ class RleDecoder {
     DCHECK_LE(bit_width_, 64);
   }
 
-  RleDecoder() {}
+  RleDecoder() : bit_width_(-1) {}
 
-  // Gets the next value.  Returns false if there are no more.
+  void Reset(const uint8_t* buffer, int buffer_len, int bit_width) {
+    DCHECK_GE(bit_width, 0);
+    DCHECK_LE(bit_width, 64);
+    bit_reader_.Reset(buffer, buffer_len);
+    bit_width_ = bit_width;
+    current_value_ = 0;
+    repeat_count_ = 0;
+    literal_count_ = 0;
+  }
+
+  /// Gets the next value.  Returns false if there are no more.
   template<typename T>
   bool Get(T* val);
 
- private:
+ protected:
```

Comment:
Since this was a copy paste job I would rather not make any unnecessary changes -- I'm also exploring the possibility of making parquet-cpp a dependency of Impala and this would cause code breakage.